### ISSUE TITLE
Site is not cached in Cloudflare

### ISSUE THREAD

**Issue by hunghvu** [hunghvu/hungvu.tech#98]
![image](https://github.com/hunghvu/hungvu.tech/assets/62678592/29f3649f-4caf-4d56-822e-4ec125967303)

```
# Response header
cache-control: private, no-cache, no-store, max-age=0, must-revalidate
cf-cache-status: BYPASS
```

It seems the page bypass Cloudflare cache, resulting in a very low cache hit. 
- This might be due to unexpected Next.js cache header setting, need to investigate.
- This might be due to the default behavior of Cloudflare, and the link below is worth looking into:

- https://community.cloudflare.com/t/doesnt-seem-to-be-caching/502969/5
- https://developers.cloudflare.com/cache/how-to/edge-browser-cache-ttl/create-page-rules/#cache-everything

**Comment by hunghvu** [comment]
Added these rules. We will monitor for a few days to observe the behavior.

![image](https://github.com/hunghvu/hungvu.tech/assets/62678592/12b78a76-d3b7-4f16-b121-b911a675b9eb)

![image](https://github.com/hunghvu/hungvu.tech/assets/62678592/ea608d2c-f583-45d8-a23f-e811e351d3ed)

**Comment by hunghvu** [comment]
These can be useful resources:
https://focusreactive.com/configure-cdn-caching-for-self-hosted-next-js-websites/
https://nextjs.org/docs/pages/building-your-application/deploying/production-checklist#caching
https://github.com/vercel/next.js/discussions/51460

**Comment by hunghvu** [comment]
After further investigation here is the observed behavior.

- HTML - bypass
- JS - cache hit
- CSS - not applicable (no CSS file)
- Fonts -  no cache-related headers
- Image (in articles, ICO) - cache hit

**Comment by hunghvu** [comment]
Since overriding the header, the cache ratio is now at a good spot.
![image](https://github.com/hunghvu/hungvu.tech/assets/62678592/916123d3-3262-4f96-9ae3-dc3b52dac476)

**Comment by hunghvu** [comment]
The cache has been working well since our fixes during v1.1 development. Two main things were implemented.

- Enforce cache header from Next.js server.
- Enforce the Edge cache rule on Cloudflare.

It is important that the status code TTL of Edge must be specified, or else the cache rule will not work.

![image](https://github.com/hunghvu/hungvu.tech/assets/62678592/0885bcda-db83-43c8-ab47-2d67a4c06748)
![image](https://github.com/hunghvu/hungvu.tech/assets/62678592/0492fba9-47a7-4673-a6be-297a9b8e5ff2)

This issue can be closed now.
